src/ORM/Marshaller.php

```diff
@@ -88,19 +88,24 @@ protected function _buildPropertyMap(array $data, array $options): array
                 $key = $nested;
                 $nested = [];
             }
+
+            $stringifiedKey = (string)$key;
             // If the key is not a special field like _ids or _joinData
             // it is a missing association that we should error on.
-            if (!$this->_table->hasAssociation((string)$key)) {
-                if (!str_starts_with((string)$key, '_')) {
+            if (!$this->_table->hasAssociation($stringifiedKey)) {
+                if (
+                    !str_starts_with($stringifiedKey, '_')
+                    && (!isset($options['junctionProperty']) || $options['junctionProperty'] !== $stringifiedKey)
```

Comment:
Not too thrilled about this. But adding the junction property name in the `Marshaller::one()` options is the only way I can make it available for this defensive coding check.

Reply:
Yeah needing to pass the extra option around is unfortunate.